### ISSUE TITLE
alphabet file

### ISSUE THREAD

**Issue by vRicardo98** [s-omranpour/ConvolutionalSpeechRecognition#3]
Hi,
when you refer to "The root directory of dataset shoud have an alphabet file", where do you get this file from and what does it contain? Cause i cannot run your version without fully understanding where does this file come from.

Best regards

**Comment by s-omranpour** [comment]
hi,
the alphabet.pkl is a python list consisting of your alphabet characters and serialized by pickle module.

```
alphabet = ['a','b','c',...,'z']
pickle.dump(alphabet, open('alphabet.pkl','wb'))
```

**Comment by vRicardo98** [comment]
Thank you very much for your reply.
If it is not too much to ask... the csv file (dataset.csv) only contains two cells, one saying "audio_pah" and another saying "sentence"? or in the cells bellow those, i need to put something else?

Best regards

**Comment by s-omranpour** [comment]
That's right. in the csv you only need the audiofile path and the corresponding sentence. nothing more.

best

**Comment by vRicardo98** [comment]
![Screenshot from 2020-05-20 09-42-22](https://user-images.githubusercontent.com/41165271/82425330-43aa1c00-9a7e-11ea-9fd5-c51651cddb50.png)

what does this error mean? It happens in function get_loaders() of dataloading.py

**Comment by vRicardo98** [comment]
from train_dataset, val_dataset, test_dataset....
only val_dataset and test_dataset are getting data for the test. the train_dataset is not getting any data from random_split().

how can i solve it? is it a problem with the dataset that i am using?
btw, i am using librispeech dev-clean, just to see if it works, than i would scale things up

thank you for help till now

**Comment by s-omranpour** [comment]
`train_loader, val_loader, test_loader, splits = get_loaders(dataset, batch_size=64, splits=[.7, 0.1, 0.2], workers=4)`

the splits argument should specify the fraction of data for each split. since the  dataset uses the .csv file as reference I assume there's sth wrong with your file. you can  check `len(dataset) > 0`to be sure about that.

**Comment by vRicardo98** [comment]
thanks for all your help! i now have your project working on my pc with the LibriSpeech dataset

**Comment by usamaehsan** [comment]
@vRicardo98 have you trained librispeech dataset on this model? If yes how much WER  you have got?

**Comment by vRicardo98** [comment]
@usamaehsan yes I have trained it with the LibriSpeech dataset that is referred in the paper related to @s-omranpour‘s work, but I only used 670 files from that dataset, because more than that would’ve take me forever to get results and I was short on time. Even though, I managed to get 5 results, in which the best one, was a WER around 14.06% with the following characteristics:
1. Train/Val/Test percentages -> 0.6/0.1/0.3
2. Batch size of 32
3. Dropout value of 0.5
4. Learning rate of 0.005
5. The array of dilations was [1,3,5]

I hope you can get better results! Anything else I can help you with, just say,.
Regards

**Comment by usamaehsan** [comment]
@vRicardo98 thanks for your reply, I really appreciate it.
Have you tried any other speech recognition system in which you have got better result than this?

**Comment by vRicardo98** [comment]
This was a research done by me and a colleague of mine, he used this https://github.com/DemisEom/SpecAugment and managed to obtain an accuracy around 0.11 +/- 0.05. This result was a bit desapointing, but not always we would obtain results and when we did, they would be like this one...

**Comment by hieuhv94** [comment]
> @usamaehsan yes I have trained it with the LibriSpeech dataset that is referred in the paper related to @s-omranpour‘s work, but I only used 670 files from that dataset, because more than that would’ve take me forever to get results and I was short on time. Even though, I managed to get 5 results, in which the best one, was a WER around 14.06% with the following characteristics:
> 
> 1. Train/Val/Test percentages -> 0.6/0.1/0.3
> 2. Batch size of 32
> 3. Dropout value of 0.5
> 4. Learning rate of 0.005
> 5. The array of dilations was [1,3,5]
> 
> I hope you can get better results! Anything else I can help you with, just say,.
> Regards

Hi @vRicardo98 !
I train this model with data-clean-100hr of librispeech dataset but it is failed.
Can u share me params of model you used, now I use default params from source of s-omranpour

This is my results after 100 first epochs:
epoch 10  train-loss : 43.53755154225843  val-loss : 43.317366296853095
epoch 20  train-loss : 43.51183282426705  val-loss : 42.26231483221472
epoch 30  train-loss : 44.06832050236407  val-loss : 42.04413370212003
epoch 40  train-loss : 43.683474623588836  val-loss : 42.18577103425944
epoch 50  train-loss : 43.419794641020395  val-loss : 41.84158054569751
epoch 60  train-loss : 43.02631453337266  val-loss : 41.450227787567364
epoch 70  train-loss : 44.03663617532445  val-loss : 42.15577789210371
epoch 80  train-loss : 43.35859699730497  val-loss : 42.826464503596135
epoch 90  train-loss : 43.261045736267796  val-loss : 41.89577377431651
epoch 100  train-loss : 43.1241825792758  val-loss : 41.75998310954204

Thank you!!
